### PR TITLE
added support for MX_ALT_OUTPUT_ROOT

### PR DESCRIPTION
**Pull request by dougxc**
This addresses #75.

Note that an mx global option cannot be used for this as some suites want their output root before command line options have been parsed.

### CODE REVIEW

**Comment by zapster** [comment]
looks good to me :+1:

File: mx.py [hunk 1]
```diff
@@ -9567,17 +9575,36 @@ def processDep(dep, edge):
...
     # Changes to these values should cause regeneration of the project files.
-    value = 'logToConsole=%s\n' % logToConsole
-    if os.environ.get('PATH'):
-        value = value + 'PATH=%s\n' % os.environ['PATH']
-    if os.environ.get('JAVA_HOME'):
-        value = value + 'JAVA_HOME=%s\n' % os.environ['JAVA_HOME']
-    if os.environ.get('DEFAULT_VM'):
-        value = value + 'DEFAULT_VM=%s\n' % os.environ['DEFAULT_VM']
-    return value
+    settings = 'logToConsole=%s\n' % logToConsole
```
Comment (dougxc) [inline mx.py]: @tkrodriguez , please check that this is semantically equivalent to how it was before my change.

**Comment by juergenchrist** [comment]
Looks good to me.

File: mx.py (hunk 1 above)
Comment (tkrodriguez) [inline mx.py]: looks good

**Pull request merged**